packages/canvas/canvas-mesh/src/Mesh.ts

```diff
@@ -19,7 +21,15 @@ Mesh.prototype._renderCanvas = function _renderCanvas(renderer: CanvasRenderer):
         this.calculateUvs();
     }
 
-    this.material._renderCanvas(renderer, this);
+    if (this.material._renderCanvas)
+    {
+        this.material._renderCanvas(renderer, this);
+    }
+    else if (!warned)
+    {
+        warned = true;
+        console.warn('Mesh is not supported in canvasRenderer.');
```

Comment:
We need a window.console check for for running in IE9. And tweaked the message a little.
```suggestion
        if (window.console) 
        {
            console.warn('Mesh with custom shaders are not supported in CanvasRenderer.');
        }
```